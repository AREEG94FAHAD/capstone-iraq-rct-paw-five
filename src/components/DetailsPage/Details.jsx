import React from 'react'
import { Row, Container, Col } from 'react-bootstrap'
import { useTranslation } from 'react-i18next'
import { PropTypes } from 'prop-types'
import { useSelector} from 'react-redux'
import 'bootstrap/dist/css/bootstrap.min.css'
import '../../App.css'
import im2 from '../../images/details-page/Rectangle 79.svg'
import im3 from '../../images/details-page/Rectangle 80.svg'
import im4 from '../../images/details-page/image 15.svg'

function Detail({match}) {
  const { t } = useTranslation()
  const { currentPets} = useSelector(
    state => state.pets
  )
  const petInfo=currentPets.find(pet=>pet.petName===match.params.petName)
  return (
    <Container fluid>
      <Row>
        <Col>
          <h2 className="row justify-content-center pt-5 pb-5 " id="colorset1">
           {t('detalis-page.PetDetalis')}
          </h2>
        </Col>
      </Row>
  
      <div> <Row className=" pr-3 pb-5">
        {/** image column */}
        <Col>{}</Col>
        <Col lg={5} sm={12} md={12} xs={12} className="pt-3 ">
          {' '}
          <img src={petInfo.file} alt="pet" className="main-image img-fluid mx-2" />
        </Col>
        {/** image column */}

        <Col lg={5} className="pb-3">
          <h3 className="pb-2 pt-3 tips text-center text-lg-left" id="colorset1">
            {petInfo.petName}
          </h3>
          <Col lg={8} md={12} sm={12}>
          <div className="pt-3">
            <label htmlFor="pet species" className="label" id="colorset">
            {t('detalis-page.pet-species')}
            </label>
            <span className="pl-2 pr-5 ">
              {petInfo.species}
            </span>
          </div>
          <div className="pt-3">
            <label htmlFor="pet age" className="label" id="colorset">
            {t('detalis-page.pet-age')}
            </label>
            <span className="pl-2">{petInfo.age}</span>
          </div>
          <div className="pt-3">
            <label htmlFor="pet-info" className="label" id="colorset">
            {t('detalis-page.pet-color')}
            </label>
            <span className="pl-2">{petInfo.color}</span>
          </div>
          <div className="pt-3">
            <label htmlFor="pet gender" className="label" id="colorset">
            {t('detalis-page.gender')}
            </label>
            <span className="pl-2">{petInfo.gender}</span>
          </div>
          <div className="pt-3">
            <label htmlFor="pet weight" className="label" id="colorset">
            {t('detalis-page.pet-weight')}
            </label>
            <span className="pl-2">{petInfo.weight}</span>
          </div>
          <div className="pt-3">
            <label htmlFor="pet story" className="label" id="colorset">
             {t('detalis-page.pet-story')}
            </label>
            <span className="text pl-2">{petInfo.story}</span>
          </div>
        </Col>
         
              <Col>{}</Col>
        </Col>
        <Col>{}</Col>
      </Row>
      {/** contact info */}

      <Row id="contact-header">
        <Col>
          <h2 className="row justify-content-center pt-5 pb-5 " id="colorset1">
          {t('detalis-page.contact-info')}
          </h2>
        </Col>
      </Row>
      <Row className="contactinfo pb-5 pl-3 pl-lg-0 ">
        <Col>{}</Col>

        <Col lg={8} md={12} sm={12}>
          <div>
            <label htmlFor="owner name" className="label" id="colorset">
            {t('detalis-page.ownerName')}
            </label>
            <span className="span pl-2">{petInfo.ownerName}</span>
          </div>
          <div className="pt-3">
            <label htmlFor="about owner" className="label" id="colorset">
            {t('detalis-page.aboutOwner')}
            </label>
            <span className="pl-2 pr-5 ">
              {petInfo.aboutOwner}
            </span>
          </div>
          <div className="pt-3">
            <label htmlFor="city" className="label" id="colorset">
            {t('detalis-page.city')}
            </label>
            <span className="pl-2">{petInfo.city}</span>
          </div>
          <div className="pt-3">
            <label htmlFor="address" className="label" id="colorset">
            {t('detalis-page.address')}
            </label>
            <span className="pl-2">{petInfo.address}</span>
          </div>
          <div className="pt-3">
            <label htmlFor="email" className="label" id="colorset">
            {t('detalis-page.email')}
            </label>
            <span className="pl-2">{petInfo.email}</span>
          </div>
          <div className="pt-3">
            <label htmlFor="phone number" className="label" id="colorset">
            {t('detalis-page.phoneNumber')}
            </label>
            <span className="pl-2">{petInfo.phoneNumber}</span>
          </div>
        </Col>
        <Col lg={3}>{}</Col>
      </Row>
      {/** pet gallery */}
      <Row className="pet-gallery">
        <Col>
          <h2
            style={{ color: 'white' }}
            className=" row justify-content-center pt-3 pb-3 "
          >
           {t('detalis-page.petGallery')}
          </h2>
        </Col>
      </Row>

      <Row className="pet-gallery pt-1 pb-5 ">
        <Col>
          <img src={im2} sm={4} className="img-fluid img pt-4 " alt="" />
        </Col>

        <Col sm={4}>
          <img src={im3} className="img-fluid img pt-4 " alt="" />
        </Col>

        <Col sm={4}>
          <img src={im4} className="img-fluid img pt-4 " alt="" />
        </Col>
      </Row>
      </div> 
   
    </Container>
  )
}

Detail.propTypes = {
  match: PropTypes.arrayOf,
}

Detail.defaultProps = {
  match: {},
}

export default Detail
