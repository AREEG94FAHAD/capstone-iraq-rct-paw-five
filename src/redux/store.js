import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import logger from 'redux-logger'
import combineReducers from './reducers/AllReducers'

const store = createStore(
  combineReducers,
  composeWithDevTools(applyMiddleware(logger))
)

export default store
