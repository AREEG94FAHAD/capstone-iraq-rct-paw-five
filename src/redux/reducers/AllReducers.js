import { combineReducers } from 'redux'
import petReducer, {
  langReducer,
  myPets,
  deleteMyPet,
  updateMyPet,
} from './petReducer'
import user from './user'

export default combineReducers({
  langReducer,
  pets: petReducer,
  user,
  myPets,
  deleteMyPet,
  updateMyPet,
})
