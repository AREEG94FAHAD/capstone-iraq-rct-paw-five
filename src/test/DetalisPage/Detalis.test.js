import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { createStore } from 'redux'

import Details from '../../components/DetailsPage/Details'
import AllReducers from '../../redux/reducers/AllReducers'

const store = createStore(AllReducers)

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Details match={{ params: 'temp' }} />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
