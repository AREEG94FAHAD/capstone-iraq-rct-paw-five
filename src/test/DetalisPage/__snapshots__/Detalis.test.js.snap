// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h2
        className="row justify-content-center pt-5 pb-5 "
        id="colorset1"
      >
        detalis-page.PetDetalis
      </h2>
    </div>
  </div>
  <div>
     
    <div
      className=" pr-3 pb-5 row"
    >
      <div
        className="col"
      />
      <div
        className="pt-3  col-lg-5 col-md-12 col-sm-12 col-12"
      >
         
        
      </div>
      <div
        className="pb-3 col-lg-5"
      >
        
        <div
          className="col-lg-8 col-md-12 col-sm-12"
        >
          <div
            className="pt-3"
          >
            <label
              className="label"
              htmlFor="pet species"
              id="colorset"
            >
              detalis-page.pet-species
            </label>
            
          </div>
          <div
            className="pt-3"
          >
            <label
              className="label"
              htmlFor="pet age"
              id="colorset"
            >
              detalis-page.pet-age
            </label>
            
          </div>
          <div
            className="pt-3"
          >
            <label
              className="label"
              htmlFor="pet-info"
              id="colorset"
            >
              detalis-page.pet-color
            </label>
            
          </div>
          <div
            className="pt-3"
          >
            <label
              className="label"
              htmlFor="pet gender"
              id="colorset"
            >
              detalis-page.gender
            </label>
            
          </div>
          <div
            className="pt-3"
          >
            <label
              className="label"
              htmlFor="pet weight"
              id="colorset"
            >
              detalis-page.pet-weight
            </label>
            
          </div>
          <div
            className="pt-3"
          >
            <label
              className="label"
              htmlFor="pet story"
              id="colorset"
            >
              detalis-page.pet-story
            </label>
            
          </div>
        </div>
        <div
          className="col"
        />
      </div>
      <div
        className="col"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h2
          className="row justify-content-center pt-5 pb-5 "
          id="colorset1"
        >
          detalis-page.health-info
        </h2>
      </div>
    </div>
    <div
      className="pb-5 pl-3 pl-lg-0  row"
    >
      <div
        className="col"
      />
      <div
        className="col-lg-8 col-md-12 col-sm-12"
      >
        <div>
          <label
            className="label"
            htmlFor="pet health"
            id="colorset"
          >
            detalis-page.petHealth
          </label>
          
        </div>
        <div
          className="pt-3"
        >
          <label
            className="label"
            htmlFor="vaccination status"
            id="colorset"
          >
            detalis-page.vaccinationStatus
          </label>
          
        </div>
        <div
          className="pt-3"
        >
          <label
            className="label"
            htmlFor="fertilization Status"
            id="colorset"
          >
            detalis-page.fertilizationStatus
          </label>
          
        </div>
        <div
          className="pt-3"
        >
          <label
            className="label"
            htmlFor="deworming Status"
            id="colorset"
          >
            detalis-page.dewormingStatus
          </label>
          
        </div>
      </div>
      <div
        className="col-lg-3"
      />
    </div>
    <div
      className="row"
      id="contact-header"
    >
      <div
        className="col"
      >
        <h2
          className="row justify-content-center pt-5 pb-5 "
          id="colorset1"
        >
          detalis-page.contact-info
        </h2>
      </div>
    </div>
    <div
      className="contactinfo pb-5 pl-3 pl-lg-0  row"
    >
      <div
        className="col"
      />
      <div
        className="col-lg-8 col-md-12 col-sm-12"
      >
        <div>
          <label
            className="label"
            htmlFor="owner name"
            id="colorset"
          >
            detalis-page.ownerName
          </label>
          
        </div>
        <div
          className="pt-3"
        >
          <label
            className="label"
            htmlFor="about owner"
            id="colorset"
          >
            detalis-page.aboutOwner
          </label>
          
        </div>
        <div
          className="pt-3"
        >
          <label
            className="label"
            htmlFor="city"
            id="colorset"
          >
            detalis-page.city
          </label>
          
        </div>
        <div
          className="pt-3"
        >
          <label
            className="label"
            htmlFor="address"
            id="colorset"
          >
            detalis-page.address
          </label>
          
        </div>
        <div
          className="pt-3"
        >
          <label
            className="label"
            htmlFor="email"
            id="colorset"
          >
            detalis-page.email
          </label>
          
        </div>
        <div
          className="pt-3"
        >
          <label
            className="label"
            htmlFor="phone number"
            id="colorset"
          >
            detalis-page.phoneNumber
          </label>
          
        </div>
      </div>
      <div
        className="col-lg-3"
      />
    </div>
    <div
      className="d-flex justify-content-center container-fluid"
    >
      <div
        className="w-100"
      >
        <p
          className="h2 text-danger p-3"
        >
          detalis-page.comments
        </p>
        <div>
          
        </div>
      </div>
    </div>
    <div
      className="pet-gallery row"
    >
      <div
        className="col"
      >
        <h2
          className=" row justify-content-center pt-3 pb-3 "
          style={
            Object {
              "color": "white",
            }
          }
        >
          detalis-page.petGallery
        </h2>
      </div>
    </div>
    <div
      className="pet-gallery pt-1 pb-5  row"
    >
      <div
        className="col"
      >
        <img
          alt=""
          className="img-fluid img pt-4 "
          sm={4}
          src="Rectangle 79.svg"
        />
      </div>
      <div
        className="col-sm-4"
      >
        <img
          alt=""
          className="img-fluid img pt-4 "
          src="Rectangle 80.svg"
        />
      </div>
      <div
        className="col-sm-4"
      >
        <img
          alt=""
          className="img-fluid img pt-4 "
          src="image 15.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
